.github/workflows/security.yml

```diff
@@ -0,0 +1,44 @@
+name: Security
+on:
+  push:
+    tags-ignore:
+      - "**"
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  security-scan:
+    name: security-scan
+    strategy:
+      matrix:
+        go: ['1.20']
+        os: [ubuntu-latest]
+      fail-fast: true
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Go ${{ matrix.go }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go }}
+          cache: false
+
+      - name: Run gosec
+        if: matrix.os == 'ubuntu-latest'
+        uses: securego/gosec@master
+        with:
+          args: ./...
```

Comment:
I suggest to run golangcilint like in https://github.com/shipwright-io/build/blob/main/.github/workflows/verify.yaml#L33, with a configuration that enables gosec like in https://github.com/shipwright-io/build/blob/main/.golangci.yaml.

Reply:
I would argue that it would be better to keep the security related items in a single yaml file and run them independently of the golangci-lint, as from what I can see, probably 90% or more of the `linters` that it runs are not security related, and I am concerned that security issues could get lost in the noise. We can also independently version gosec if needed.